xarray/core/utils.py

```diff
@@ -38,6 +37,18 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _maybe_cast_to_netcdftimeindex(index):
+    try:
+        from netcdftime._netcdftime import datetime as ncdatetime
+        from ..conventions.netcdftimeindex import NetCDFTimeIndex
+        if len(index):
+            if isinstance(index[0], ncdatetime):
```

Comment:
Here and in other places (e.g. [here](https://github.com/pydata/xarray/pull/1252/commits/a177f89d1f4f63fc58040f4a4f68e175407a67c4#diff-796786eab6c7f79e802c9f4617ef7059R603), and [here](https://github.com/pydata/xarray/pull/1252/commits/a177f89d1f4f63fc58040f4a4f68e175407a67c4#diff-796786eab6c7f79e802c9f4617ef7059R201)) I need to check if an array contains netcdftime objects (or check some property of those netcdftime objects).  Since just querying `array.dtype` returns the generic `'O'` in this case, I need to look inside the array to get this information.  Is there possibly a cleaner way of doing this?